applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_yield_surface.cpp

```diff
@@ -0,0 +1,71 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Mohamed Nabi
+//
+
+#include "custom_constitutive/coulomb_yield_surface.h"
+#include "custom_constitutive/tension_cutoff.h"
+#include "tests/cpp_tests/geo_mechanics_fast_suite.h"
+#include "tests/cpp_tests/test_utilities.h"
+#include "utilities/math_utils.h"
+
+#include <boost/numeric/ublas/assignment.hpp>
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(TestCoulombYieldSurface, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    double           friction_angle = MathUtils<>::DegreesToRadians(45.0);
+    constexpr double cohesion       = 2.0;
+
+    CoulombYieldSurface coulombYieldSurface(friction_angle, cohesion, 0.0);
+
+    Vector principal_stress(3);
+    principal_stress <<= 3.0, 2.0, 1.0;
+    double yield_value       = coulombYieldSurface.YieldFunctionValue(principal_stress);
+    double expected_solution = 1.0;
+    KRATOS_EXPECT_NEAR(yield_value, expected_solution, Defaults::absolute_tolerance);
```

Comment:
We don't need to introduce these variables in my opinion. Using their values directly is just as clear and reduces the number of variables needed:
```suggestion
    KRATOS_EXPECT_NEAR(coulombYieldSurface.YieldFunctionValue(principal_stress), 1.0, Defaults::absolute_tolerance);
```